pkg/ast/spl/spl.peg

```diff
@@ -2065,6 +2098,25 @@ EvalComparisonExpr <- (op:("isbool" / "isint" / "isstr" / "isnull" / "isnotnull"
         RightValue: nil,
         ValueOp: opNameStr,
     }
+
+    if opNameStr != "searchmatch" {
+        return expr, nil
+    }
+
+    valExpr := value.(*structs.ValueExpr)
+    if valExpr.StringExpr == nil {
+        return nil, fmt.Errorf("Spl peg: Searchmatch should only have string values as argument")
+    }
+
+    fieldsList, err := SearchMatchGetFields(valExpr.StringExpr.RawString)
+    if err != nil {
+        return nil, err
+    }
+
+    valExpr.StringExpr.StringExprMode = structs.SEMFieldList
+    valExpr.StringExpr.FieldList = fieldsList
+    expr.LeftValue = valExpr
```

Comment:
nit: instead of
```
    if opNameStr != "searchmatch" {
        return expr, nil
    }
```
And then handling the logic for `searchmatch`, I think it would be easier to read (especially if we have to add more special cases later) if you do
```
if opNameStr == "searchmatch" {
    // handle the logic here
}
```
Or you could move the searchmatch logic to a different function if you want to

Reply:
addressed in latest commit